### PR TITLE
chore: Add separate request/response views

### PR DESCRIPTION
**Pull request by sharat87**
This PR gets finer control into what fields are allowed in request-body-only, vs what's allowed in response-body-only. This leaves the fields to separately controlled regarding what can go into the database and what can't.

[Slack thread](https://theappsmith.slack.com/archives/CPQNLFHTN/p1710125307810949).

## Summary by CodeRabbit

- **Refactor**
	- Enhanced data handling in action controllers for improved performance and maintainability.
- **New Features**
	- Introduced interfaces for better management of data views in action responses.
- **Documentation**
	- Added documentation for new interfaces and their application.

### CODE REVIEW

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/views/ResponseOnly.java [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package com.appsmith.external.views;
+
+public interface ResponseOnly extends Views.Public {}
```
Comment (abhvsn) [inline app/server/appsmith-interfaces/src/main/java/com/appsmith/external/views/ResponseOnly.java:3]: Let's add a comment why do we need this view even if we have `Views.Public` for API responses.

**Review by abhvsn — COMMENTED** [review]
@sharat87 I can see other domains e.g. BaseDomain, LayoutDTO, DatasourceDTO, Application etc and some are present in EE repo as well. Are we going to add the RequestOnly annotation in other controller classes as well?

**Comment by sharat87** [comment]
@abhvsn, not as a singular task right now, no. I'll move the APIs that are blocking my efforts on Postgres, but other than that, I'd rather not touch what's not broken. Eventually though, I'd like to completely remove `Public` and just use `RequestOnly` and `ResponseOnly`, but that's not pressing enough to happen as a single big bang. Does that sound okay?

**Review by abhvsn — APPROVED** [review]

**Comment by abhvsn** [comment]
> @abhvsn, not as a singular task right now, no. I'll move the APIs that are blocking my efforts on Postgres, but other than that, I'd rather not touch what's not broken. Eventually though, I'd like to completely remove Public and just use RequestOnly and ResponseOnly, but that's not pressing enough to happen as a single big bang. Does that sound okay?

@sharat87 sounds good, let's create a ticket to track this request though. Approved!

**Review by mohanarpit — APPROVED** [review]

**Pull request merged**